doc/source/whatsnew/v0.18.1.txt

```diff
@@ -29,12 +29,16 @@ New features
 Enhancements
 ~~~~~~~~~~~~
 
+- ``Index.take`` now handles ``allow_fill`` and ``fill_value`` consistently (:issue:`12631`)
```

Comment:
maybe let's put a comment in the actual doc-string. This is not new, but very typical of using ``-1`` to represent missing values in indexers.